### ISSUE TITLE
Cannot set property default of #<Object> which has only a getter

### ISSUE THREAD

**Issue by bluelovers** [microsoft/TypeScript#37090]
**TypeScript Version:**  3.7.x-dev.201xxxxx

**Search Terms:** which has only a getter

**Code**

https://github.com/bluelovers/ws-react/blob/master/packages/react-use-localstorage/src/core.ts

```ts
export * from './core/v4';
import createStorageHook from './core/v4';
export default createStorageHook
```

```ts
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (!exports.hasOwnProperty(p)) __createBinding(exports, m, p);
}
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
__exportStar(require("./core/v4"), exports);
const v4_1 = __importDefault(require("./core/v4"));
exports.default = v4_1.default;
//# sourceMappingURL=core.js.map
```

**Expected behavior:**

no crash error

**Actual behavior:**

TypeError: Cannot set property default of #<Object> which has only a getter

**Playground Link:** 

**Related Issues:** 

https://github.com/microsoft/TypeScript/issues/37092

**Comment by bluelovers** [comment]
if move down `export *` after `export default` will not have this error

but i think this should not happen

```ts
import createStorageHook from './core/v4';
export default createStorageHook
export * from './core/v4';
```

Related PR:
https://github.com/microsoft/TypeScript/pull/35967

**Comment by rokoroku** [comment]
Maybe related: https://github.com/microsoft/tslib/issues/103

I went through the same issue and I solved it by downgrading tslib to 1.13.0

**Comment by rbuckton** [comment]
This was fixed by #38808.
